crates/bevy_winit/src/lib.rs

```diff
@@ -344,12 +346,26 @@ fn handle_winit_event(
     redraw_event_reader: &mut ManualEventReader<RequestRedraw>,
     winit_events: &mut Vec<WinitEvent>,
     exit_notify: &SyncSender<AppExit>,
+    exit_happened: &mut Option<AppExit>,
     event: Event<UserEvent>,
     event_loop: &EventLoopWindowTarget<UserEvent>,
 ) {
     #[cfg(feature = "trace")]
     let _span = bevy_utils::tracing::info_span!("winit event_handler").entered();
 
+    // We guarantee that after a user send a `AppExit` the game won't update again.
+    // Problem is winit will send us queued events even after we have triggered a
+    // loop exit. These events then trigger a game update.
+    //
+    // TODO: This is a bodge. A better solution would be to go and figure out
+    //       which winit events we need to handle after the app exited and letting
+    //       those through. The non-send PR will be replacing all of this so
+    //       we'll likely fix this then or a bit after.
+    if exit_happened.is_some() && event != Event::LoopExiting {
+        info!("App is exiting. Dropping event: {event:?}");
```

Comment:
```suggestion
        debug!("App is exiting. Dropping event: {event:?}");
```

I'd consider switching this to debug instead of info. If it's info, then this line gets called at least twice on `DefaultPlugins` programs. I think it's too much of an internal detail to deem explaining why it gets logged every time `AppExit` is sent.

<img width="1076" alt="image" src="https://github.com/bevyengine/bevy/assets/59022059/bb027f88-dd21-4b60-8cd6-b28b00167007">